### ISSUE TITLE
Specified Dockerfile ignored on Windows only

### ISSUE THREAD

**Issue by pageer** [lando/lando#3506]
I'm using a Lando configuration that specifies a Dockerfile, with the intention that we can use the same Dockerfile for development that we use for production.  Here is a stripped-down version of my configuration:

```
# .lando.yml

name: document-automation-test
recipe: laravel
config:
    via: nginx
    database: mysql:5.7
    webroot: public
    xdebug: 'debug,develop'

services:
    appserver:
        overrides:
            build:
                context: .
                dockerfile: ./docker/prod/Dockerfile

# docker/prod/Dockerfile

FROM php:8.1-fpm-bullseye
```

**Expected behavior:**
On all platforms, Lando should spin up the appserver and use the specified Dockerfile, i.e. the php8.1 image.

**Actual behavior:**
On Linux and MacOS, Lando spins up the appserver with the php8.1 image.  On Windows, it ignores the Dockerfile and spins up the appserver with a php7.4 image.

Lando version: 3.10.0
Docker version: 4.16.2 with WSL2 backend
Windows version: Win10 Pro, version 21H2, build 19044.2364

When Lando on Windows starts up the appserver, there are no errors.  It spins up a working environment, it just ignores the specified Dockerfile and uses the default php7.4 image.  FWIW, I also tried running the Linux build of Lando with docker-ce directly inside WSL and observed the same behavior.  When I `lando start` the same configuration on a real Ubuntu box, the appserver spins up without error and using the php8.1 image.  I looked at the files being generated in `~/.lando/compose/` and I'm not seeing any obvious differences between Windows and Linux other than the paths.

**Comment by labboy0276** [comment]
Have you tried the override like this:

`                
context: ./docker/prod
dockerfile: Dockerfile
`

**Comment by pageer** [comment]
I tried that and a few other path variations.  I still see the same behavior.

**Comment by AaronFeledy** [comment]
Check your `C:\Users\YOURNAME\.lando\compose`  for the docker-compose files that are generated by Lando. Lando basically generates docker compose files then runs a `docker compose up` on them. Perhaps there are some clues in those files like a bad dockerfile path or something.

**Comment by pageer** [comment]
I did look at the files in `~/.lando/compose`.  I'm not a Docker expert, but nothing looked obviously wrong or inconsistent with the Linux box that's working as expected.

I do see that `~/.lando/compose/documentautomation/appserver-0.yml` starts off with this:
```
services:
  appserver:
    image: 'devwithlando/php:7.4-fpm-4'
```
I guess that's where the php7.4 image is coming from.  However, the `appserver-0.yml` on the Linux box starts the same way, which leads me to believe that's not the problem.

The only place where I see the Dockerfile referenced is in `appserver-3.yml`, which just contains this:
```
services:
  appserver:
    environment:
      APP_LOG: errorlog
    build:
      context: 'C:\Users\pgeer\Projects\document-automation'
      dockerfile: ./docker/prod/Dockerfile
version: '3.6'
```
I'm not experienced with docker-compose, but I don't see anything obviously wrong there.  The path to the project root is right and the relative path to the Dockerfile is right.  I also tried variations on path formats (absolute paths, backslashes, forward slashes), but it didn't make any difference.  And I'm seeing the same file content on the Linux box that works, just with a different context path.  So that doesn't look like the problem either.

I did a diff of the `~/.lando/compose/documentautomation/appserver-*.yml` files between my Linux and Windows boxes and they were nearly identical.  The only differences that came up were in the local file paths (which are obviously different), the Xdebug configuration (which seems irrelevant), and the `command` key in `appserver-0.yml`, which is `command: docker-php-entrypoint php-fpm -R` on Windows and `command: docker-php-entrypoint php-fpm` without the `-R` on Linux (no idea if that's significant).

If there are any other config files or logs that I should look at, or any additional testing steps that might be fruitful, I'd be grateful for suggestions.

**Comment by pageer** [comment]
Well, I figured out my problem.  I got the hint from [this file](https://github.com/lando/core/blob/main/docs/services.md). Apparently I needed to specify a custom type in the appserver section of my `.lando.yml`.
```
services:
    appserver:
        type: php:custom
        overrides:
            build:
                context: .
                dockerfile: ./docker/dev/Dockerfile
```
When I do that (and clear out a few old Docker images), Lando finds the Dockerfile and uses the correct image.  It displays in Docker Desktop as `devwithlando/php:custom-fpm-4`, but it's definitely the correct image and the rules in the Dockerfile definitely run.

So for custom Dockerfiles to get picked up, it looks like the `type: php:custom` is required on Windows.  I'm not sure why it seems to be optional on Mac and Linux, but that's the behavior I'm seeing.  But either way, that fixes my problem and doesn't seem to break other platforms.
